frontend/integration-tests/views/login.view.ts

```diff
@@ -11,15 +10,11 @@ export const userDropdown = $('[data-test=user-dropdown] .pf-c-dropdown__toggle'
 
 export const selectProvider = async(provider: string) => {
   const idpLink = element(by.cssContainingText('.idp', provider));
-  while (!(await idpLink.isPresent())) {
-    await browser.get(appHost);
-    await browser.sleep(3000);
-  }
   await idpLink.click();
 };
 
 export const login = async(providerName: string, username: string, password: string) => {
-  if (providerName) {
+  if (await $('a.idp').isPresent()) {
```

Comment:
Note that this change probably should not be necessary in openshift/console

Reply:
Without this change, it can't run test with a cluster without `idp`.
It's not necessary if the test run with `idp` always, I know it's not perfect.

I have a PR in console about this: https://github.com/openshift/console/pull/2016